BUILD.md

```diff
@@ -41,9 +41,9 @@ The [minimum compiler versions][2] required are:
 
 | Compiler    | Version |
 |-------------|---------|
-| GCC         | 10      |
+| GCC         | 11      |
 | Clang       | 13      |
-| Apple Clang | 13.1.6  |
+| Apple Clang | 14.0.3  |
```

Comment:
`using enum` is [supported](https://clang.llvm.org/cxx_status.html#cxx20) in Clang 13. Let's not change that minimum requirement until we know for sure that we need it.

Reply:
ok. but is there any disadvantage to upgrading directly to 14? since the code has been working with clang 14, I assumed we could upgrade. But I don't have strong opinions about this

Reply:
It's ok for a builder to upgrade. We don't want to tell people they have to upgrade if there is no reason to upgrade. These are minimum required versions.